src/main/java/com/epam/ta/reportportal/core/analyzer/auto/impl/AnalyzerServiceImpl.java

```diff
@@ -121,13 +127,14 @@ public void runAnalyzers(Launch launch, List<Long> testItemIds, AnalyzerConfig a
    * @param analyzerConfig Analyzer config
    */
   private void analyzeItemsPartition(Launch launch, List<Long> testItemIds,
-      AnalyzerConfig analyzerConfig) {
+      AnalyzerConfig analyzerConfig, Optional<Long> previousLaunchId) {
     LOGGER.info("Start analysis of '{}' items for launch with id '{}'", testItemIds.size(),
         launch.getId());
     List<TestItem> toAnalyze = testItemRepository.findAllById(testItemIds);
     Optional<IndexLaunch> rqLaunch = launchPreparerService.prepare(launch, toAnalyze,
         analyzerConfig);
     rqLaunch.ifPresent(rq -> {
+      previousLaunchId.ifPresent(rq::setPreviousLaunchId);
```

Comment:
Can we move the whole logic of getting and setting previous launch id into launchPreparerService?

Reply:
analyzeItemsPartition run for partion of testItem, so i getting previousLaunchId in runAnalyzers one time for all partition.